app/src/main/java/ch/hikemate/app/ui/auth/CreateAccountScreen.kt

```diff
@@ -0,0 +1,150 @@
+                      backgroundColor = primaryColor,
+                  ))
+
+  var name by remember { mutableStateOf("") }
+  var email by remember { mutableStateOf("") }
+  var password by remember { mutableStateOf("") }
+  var confirmPassword by remember { mutableStateOf("") }
+
+  Column(
+      modifier =
+          Modifier.testTag(Screen.CREATE_ACCOUNT)
+              .padding(
+                  // Add padding to the sidebar padding
+                  start = 16.dp,
+                  end = 16.dp,
+                  top = 16.dp,
+              ),
+      verticalArrangement = Arrangement.spacedBy(16.dp)) {
+        BackButton(navigationActions)
+        Text(
+            stringResource(R.string.create_account_title),
+            style = TextStyle(fontWeight = FontWeight.Bold, fontSize = 32.sp),
+            modifier = Modifier.testTag(CreateAccountScreen.TEST_TAG_TITLE))
+
+        OutlinedTextField(
+            modifier = Modifier.fillMaxWidth().testTag(CreateAccountScreen.TEST_TAG_NAME_INPUT),
+            colors = inputColors,
+            value = name,
+            onValueChange = { name = it },
+            label = { Text(stringResource(R.string.create_account_name_label)) })
+
+        OutlinedTextField(
+            modifier = Modifier.fillMaxWidth().testTag(CreateAccountScreen.TEST_TAG_EMAIL_INPUT),
+            keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Email),
+            colors = inputColors,
+            value = email,
+            onValueChange = { email = it },
+            label = { Text(stringResource(R.string.create_account_email_label)) })
+
+        OutlinedTextField(
+            modifier = Modifier.fillMaxWidth().testTag(CreateAccountScreen.TEST_TAG_PASSWORD_INPUT),
+            visualTransformation = PasswordVisualTransformation(),
+            colors = inputColors,
+            value = password,
+            onValueChange = { password = it },
+            label = { Text(stringResource(R.string.create_account_password_label)) })
+
+        OutlinedTextField(
+            modifier =
+                Modifier.fillMaxWidth()
+                    .testTag(CreateAccountScreen.TEST_TAG_CONFIRM_PASSWORD_INPUT),
+            visualTransformation = PasswordVisualTransformation(),
+            colors = inputColors,
+            value = confirmPassword,
+            onValueChange = { confirmPassword = it },
+            label = { Text(stringResource(R.string.create_account_repeat_password_label)) })
+
+        BigButton(
+            modifier = Modifier.fillMaxWidth().testTag(CreateAccountScreen.TEST_TAG_SIGN_UP_BUTTON),
+            buttonType = ButtonType.PRIMARY,
+            label = stringResource(R.string.create_account_create_account_button),
+            onClick = {
+              if (password != confirmPassword) {
+                // Show an error message in a toast
+                Toast.makeText(context, mismatchErrorMessage, Toast.LENGTH_SHORT).show()
```

Comment:
Nitpick: Since you do some checks on the password here, might be a good idea to just add an empty field check and a (quick) email check too. You can use this Regex for it:  `/^[\w-\.]+@([\w-]+\.)+[\w-]{2,4}$/`

Reply:
Nice idea. Thanks!